viamrobotics/api
# resolve dependabot, upgrade buf, lots of automatic linter changes, fix flappy stubs

## What changed
- bump x/net and protobuf libraries for CVEs
- bump buf because our version was incompatible with newer protobuf
- lots of automated changes from bumping buf -- the only manual changes are go.mod, go.sum, proto/viam/buf.yaml, tools/tools.go, and workflows in .github/
## Why
Dependabot 19, 20, 21
## Successful test run
CI appears to be failing because the workflow change is not on the main branch. There's a successful run here https://github.com/viamrobotics/api/actions/runs/11731674159.
## Manual test
This ended up being a large change because something rewrote the proto files; as cross check:
- [ ] run RDK + backend against this API branch, make sure basic things still work
## Bonus
Looks like the automatically generate stubs no longer flap!
